src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -35,296 +35,650 @@
-    * @param listContactsChangedLastName list of contacts whose lastnames are changed
-    * @return list<Account>
+    * @description The Household Account record type.
     */
-    public static list<Account> updateAdminAccountsName(List<Contact> listContactsChangedLastName) {
-        List<Account> listAccToUpdate = new List<Account>();
-        Map<Id, Contact> mapAccIdToCon = new Map<Id, Contact>();
-        for (Contact con : listContactsChangedLastName) {
-            mapAccIdToCon.put(con.AccountId, con);
+    private static ID hhAccountRecordTypeId = UTIL_Describe.getHhAccRecTypeID();
+
+    /*******************************************************************************************************
+    * @description The Administrative Account record type.
+    */
+    private static ID adminAccountRecordTypeId = UTIL_Describe.getAdminAccRecTypeID();
+
+    /*******************************************************************************************************
+    * @description The default/selected Household Account name format.
+    */
+    private static String hhNameFormat = UTIL_CustomSettingsFacade.getSettings().Household_Account_Naming_Format__c;
+
+    /*******************************************************************************************************
+    * @description The default/selected Administrative Account name format.
+    */
+    private static String aaNameFormat = UTIL_CustomSettingsFacade.getSettings().Admin_Account_Naming_Format__c;
+
+    /*******************************************************************************************************
+    * @description Update admin Account names if any changes made to Contacts' lastname.
+    * @param contactsChangedLastName list of Contacts whose LastNames have changed.
+    * @return List<Account>.
+    */
+    public static List<Account> updateAdminAccountsName(List<Contact> contactsChangedLastName) {
+        List<Account> accountsToUpdate = new List<Account>();
+        Map<Id, List<Contact>> accountIdPrimaryContactId = new Map<Id, List<Contact>>();
+
+        for (Contact con : contactsChangedLastName) {
+            if (accountIdPrimaryContactId.containsKey(con.AccountId)) {
+                accountIdPrimaryContactId.get(con.AccountId).add(con);
+            } else {
+                accountIdPrimaryContactId.put(con.AccountId, new List<Contact> {con});
+            }
+
         }
 
-        List<Account> listAcc = [SELECT Id, RecordTypeId
-                                FROM Account
-                                WHERE Id IN :mapAccIdToCon.keySet()];
-        for (Account acc : listAcc) {
-            // only consider Admin Accounts
-            if (acc.RecordTypeId == userDefinedAdminRecordTypeId) {
-                acc.Name = UTIL_ACCT_Naming.strNameAdmAccountForContact(mapAccIdToCon.get(acc.Id));
-                listAccToUpdate.add(acc);
+        for (Account acc: [SELECT Id, Name, RecordTypeId,
```

Comment:
nice moving this to the inline for -- I hate the look but they're more efficient in large sets of data +1